src/coreclr/src/jit/gentree.cpp

```diff
@@ -10869,13 +10905,14 @@ void Compiler::gtDispConst(GenTree* tree)
             if (tree->IsIconHandle(GTF_ICON_STR_HDL))
             {
                 const WCHAR* str = eeGetCPString(tree->AsIntCon()->gtIconVal);
-                if (str != nullptr)
+                // If *str points to a '\0' then don't print the string's values
+                if ((str != nullptr) && (*str != '\0'))
                 {
                     printf(" 0x%X \"%S\"", dspPtr(tree->AsIntCon()->gtIconVal), str);
                 }
-                else
+                else // We can't print the value of the string
                 {
-                    // Note that eGetCPString isn't currently implemented on Linux/ARM
+                    // Note that eeGetCPString isn't currently implemented on Linux/ARM
```

Comment:
Do we know why it's not implemented? Would it be simple to fix?

Reply:
I am not sure how useful this EE method actually is 
In my usage we rarely end up printing the actual string.
Currently it can't ever be printed at crossgen time, only at JIT time can we print the string.

Prior to my change we would try to print every string as the empty string 
We get a null terminated string, and then print it as as ""